aten/src/ATen/native/DilatedMaxPool2d.cpp

```diff
@@ -58,12 +50,83 @@ void max_pool2d_with_indices_out_cpu_template(
   const int64_t outputHeight = pooling_output_shape<int64_t>(inputHeight, kH, padH, dH, dilationH, ceil_mode);
   const int64_t outputWidth = pooling_output_shape<int64_t>(inputWidth, kW, padW, dW, dilationW, ceil_mode);
 
+  const auto memory_format = input.suggest_memory_format();
   pool2d_shape_check(
     input,
     kH, kW, dH, dW, padH, padW, dilationH, dilationW,
     nInputPlane,
     inputHeight, inputWidth,
-    outputHeight, outputWidth, input.suggest_memory_format());
+    outputHeight, outputWidth, memory_format);
+
+  /* resize output and indices */
+  if (input.ndimension() == 3) {
+    if (memory_format == at::MemoryFormat::Contiguous) {
+      set_output(0, {nInputPlane, outputHeight, outputWidth}, {outputHeight * outputWidth, outputWidth, 1}, input.options(), input.names());
+      /* indices will contain the locations for each output point */
+      set_output(1, {nInputPlane, outputHeight, outputWidth}, {outputHeight * outputWidth, outputWidth, 1}, input.options().dtype(kLong), input.names());
+    } else if (memory_format == at::MemoryFormat::ChannelsLast) {
+      set_output(0, {nInputPlane, outputHeight, outputWidth}, {1, outputWidth * nInputPlane, nInputPlane}, input.options(), input.names());
+      /* indices will contain the locations for each output point */
+      set_output(1, {nInputPlane, outputHeight, outputWidth}, {1, outputWidth * nInputPlane, nInputPlane}, input.options().dtype(kLong), input.names());
+    } else {
+      TORCH_CHECK(false, "Unsupported memory format. Supports only ChannelsLast, Contiguous");
+    }
+  } else {
+    if (memory_format == at::MemoryFormat::Contiguous) {
+      set_output(0, {nbatch, nInputPlane, outputHeight, outputWidth}, {nInputPlane * outputHeight * outputWidth, outputHeight * outputWidth, outputWidth, 1}, input.options(), input.names());
+      /* indices will contain the locations for each output point */
+      set_output(1, {nbatch, nInputPlane, outputHeight, outputWidth}, {nInputPlane * outputHeight * outputWidth, outputHeight * outputWidth, outputWidth, 1}, input.options().dtype(kLong), input.names());
+    } else if (memory_format == at::MemoryFormat::ChannelsLast) {
+      set_output(0, {nbatch, nInputPlane, outputHeight, outputWidth}, {nInputPlane * outputHeight * outputWidth, 1, outputWidth * nInputPlane, nInputPlane}, input.options(), input.names());
+      /* indices will contain the locations for each output point */
+      set_output(1, {nbatch, nInputPlane, outputHeight, outputWidth}, {nInputPlane * outputHeight * outputWidth, 1, outputWidth * nInputPlane, nInputPlane}, input.options().dtype(kLong), input.names());
```

Comment:
For the logic here, I expected the corresponding `resize_` logic in the original kernels to have been deleted. But this does not appear to have happened.

Reply:
ok, I will remove it.